### PR TITLE
Autodesk: Upgrade OpenImageIO to 2.3.15 + update unit test

### PR DESCRIPTION
**Pull request by erikaharrison-adsk**
### Description of Change(s)

Upgrade OpenImageIO to 2.3.15 and need update one exr image to make sure test case testUsdImgingGLCards passed.

Background: After OIIO upgrade start from 2.2.4, the attribute "worldtoNDC" is not mapped with "worldtoscreen" attribute in OIIO. The test case testUsdImgingGLCards needs revised exr file with legacy attribute name "worldtoscreen" that is supported in HioOiio plugin.

Note: The unit test fix is also part of https://github.com/PixarAnimationStudios/OpenUSD/pull/2603 but we need it for the OIIO here too.

### Fixes Issue(s)
- https://github.com/PixarAnimationStudios/OpenUSD/issues/2601 

- [X] I have verified that all unit tests pass with the proposed changes

- [X] I have submitted a signed Contributor License Agreement

### CODE REVIEW

**Comment by jesschimein** [comment]
Filed as internal issue #USD-8607

File: build_scripts/build_usd.py [hunk 1]
```diff
@@ -1264,11 +1264,7 @@ def InstallOpenVDB(context, force, buildArgs):
 ############################################################
 # OpenImageIO
 
-if MacOS():
```
Comment (tallytalwar) [inline build_scripts/build_usd.py:1267]: Hi @erikaharrison-adsk 

Noting that in the current dev branch and in upcoming 23.11 release, we have already unified this and moved OIIO to 2.3.21. https://github.com/PixarAnimationStudios/OpenUSD/blob/dev/build_scripts/build_usd.py#L1247

**Review by tallytalwar — COMMENTED** [review]
Hi @erikaharrison-adsk 

Since we are already on page to update our OIIO dependency to 2.3.21, I will go and close this PR. Thanks a bunch.

**Comment by erikaharrison-adsk** [comment]
Thanks so much! Looks like your changes have also addressed the unit test impact. Thank you for that too!

**Pull request closed without merging**